src/domains/profile/pages/CreateProfile/CreateProfile.tsx

```diff
@@ -61,13 +64,26 @@ export const CreateProfile = ({ onSubmit }: CreateProfileProps) => {
 				<div className="flex flex-row items-center justify-between">
 					<span className="mt-2 text-sm text-theme-neutral-dark">Want to set the wallet to dark mode?</span>
 					<div className="-mt-8">
-						<Toggle />
+						<Toggle ref={register()} name="isDarkMode" />
 					</div>
 				</div>
 			),
 		},
 	];
 
+	const submitForm = async ({ name, currency, isDarkMode, marketProvider }: any) => {
+		const profile = env.profiles().create(name);
+		profile.settings().set(ProfileSetting.MarketProvider, marketProvider);
+		profile.settings().set(ProfileSetting.ChartCurrency, currency);
+		profile.settings().set(ProfileSetting.Theme, isDarkMode ? "dark" : "light");
+
+		await env.persist();
+
+		if (onSubmit) {
```

Comment:
Why is that check needed, shouldn't it always be defined because something would go wrong if it isn't?

Reply:
Agreed, that what I was talking when I questioned about the `typeof`, ideally the submit prop shouldn't be required?

Reply:
It's removed now